certchain/certmanager/multicert_disk_cache.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package certmanager
+
+import (
+	"bytes"
+	"errors"
+	"path"
+
+	"github.com/gofrs/flock"
+	"github.com/google/renameio"
+	"github.com/google/webpackager/certchain"
+	"github.com/google/webpackager/certchain/certchainutil"
+	"github.com/hashicorp/go-multierror"
+)
+
+var (
+	errEmptyCertDir      = errors.New("certmanager: empty CertDir")
+	errEmptyCertFilename = errors.New("certmanager: empty CertFilename")
+	errEmptyOCSPFilename = errors.New("certmanager: empty OCSPFilename")
+)
+
+// MultiCertDiskCache is a Cache on a local filesystem. It writes the
+// certificate chain in the PEM format and the OCSP response in the DER format
+// to separate files as specified by MultiCertDiskCacheConfig. It uses the
+// digest of the certificate chain as the base filename for the cert and OCSP
+// files.
+type MultiCertDiskCache struct {
```

Comment:
I think we can replace the existing DiskCache or maybe rename the existing one to SingleCertDiskCache. I presume the multi-cert cache to become the default choice so want to give a shorter name to this one. I think webpackager is not yet at the state we need to keep the backward-compatibility at any cost (README.md states that API can change).

Reply:
My preference is actually to keep the names MultiCertDiskCache and also renamed DiskCache to SingleCertDiskCache, so it's clear what the disk caches are actually doing. I don't think we use these classes prevalently and naming something DiskCache is a bit more abstract and a little harder to read. I don't feel strongly about it though.
For this comment, I will rename DiskCache to SingleCertDiskCache and change the default to MultiCertDiskCache.